File: app/src/main/java/com/kimikevin/eatsplorer/model/entity/PlaceDetailsResponse.java
package com.kimikevin.eatsplorer.model.entity;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PlaceDetailsResponse {
    @SerializedName("nationalPhoneNumber")
    private String nationalPhoneNumber;

    @SerializedName("websiteUri")
    private String websiteUri;

    @SerializedName("regularOpeningHours")
    private OpeningHours regularOpeningHours;

    public String getNationalPhoneNumber() {
        return nationalPhoneNumber;
    }

    public String getWebsiteUri() {
        return websiteUri;
    }

    public OpeningHours getRegularOpeningHours() {
        return regularOpeningHours;
    }

    public boolean isOpenNow() {
        return regularOpeningHours != null && regularOpeningHours.openNow;
    }

    public static class OpeningHours {
        @SerializedName("openNow")
        public boolean openNow;
        @SerializedName("weekdayDescriptions")
        public List<String> weekdayDescriptions;
    }
}

File: app/src/main/java/com/kimikevin/eatsplorer/model/entity/PlaceDetailsResponse.kt
package com.kimikevin.eatsplorer.model.entity

data class PlaceDetailsResponse(
    val nationalPhoneNumber: String?,
    val websiteUri: String?,
    val regularOpeningHours: OpeningHours?
) {
    val isOpenNow: Boolean get() = regularOpeningHours?.openNow ?: false

    data class OpeningHours(
        val openNow: Boolean,
        val weekdayDescriptions: List<String>?
    )
}
